### ISSUE TITLE
[Bug] Server refuses to launch on 21.0.0 for Fabric 1.21 0.15.11

### ISSUE THREAD

**Issue by DarkDevero** [quiqueck/BCLib#177]
### What happened?

Launching a fabric server with betternether, bclib, and worldweaver installed on Fabric 1.21 0.15.11 crashes the server due to the following error.

### BCLib

21.0.0

### Fabric API

0.100.3

### Fabric Loader

0.15.11

### Minecraft

Older

### Relevant log output

```shell
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'bclib'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.15.11.jar:?]
	at net.minecraft.server.Main.main(Main.java:111) [server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [fabric-server-mc.1.21-loader.0.15.11-launcher.1.0.1.jar:1.0.1]
	Suppressed: java.lang.RuntimeException: Cannot load class org.betterx.wover.ui.impl.client.VersionCheckerClient in environment type SERVER
		at net.fabricmc.loader.impl.transformer.FabricTransformer.transform(FabricTransformer.java:59) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray(KnotClassDelegate.java:462) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:415) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.15.11.jar:?]
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
		at org.betterx.wover.entrypoint.Wover.onInitialize(Wover.java:14) ~[world-weaver-21.0.1.jar:?]
		at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.15.11.jar:?]
		at net.minecraft.server.Main.main(Main.java:111) [server-intermediary.jar:?]
		at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.15.11.jar:?]
		at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.15.11.jar:?]
		at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) [fabric-server-mc.1.21-loader.0.15.11-launcher.1.0.1.jar:1.0.1]
Caused by: java.lang.RuntimeException: Cannot load class org.betterx.wover.ui.impl.client.VersionCheckerClient in environment type SERVER
	at net.fabricmc.loader.impl.transformer.FabricTransformer.transform(FabricTransformer.java:59) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray(KnotClassDelegate.java:462) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:415) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.15.11.jar:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
	at org.betterx.bclib.BCLib.onInitialize(BCLib.java:71) ~[bclib-21.0.0.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384) ~[fabric-loader-0.15.11.jar:?]
	... 7 more
```

### Other Mods

```shell
appleskin-fabric-mc1.21-3.0.2.jar
architectury-13.0.1-fabric.jar
badpackets-fabric-0.8.1.jar
balm-fabric-1.21-21.0.6.jar
bclib-21.0.0.jar
better-nether-21.0.0.jar
BetterF3-11.0.1-Fabric-1.21.jar
cloth-config-15.0.127-fabric.jar
collective-1.21.0-7.61.jar
fabric-api-0.100.3+1.21.jar
fabric-carpet-1.21-1.4.147+v240613.jar
fabric-language-kotlin-1.11.0+kotlin.2.0.0.jar
ferritecore-7.0.0-fabric.jar
freecam-fabric-modrinth-1.3.0-beta1+mc1.21.jar
fullbrightnesstoggle-1.21.0-4.0.jar
ImmediatelyFast-Fabric-1.2.18+1.21.jar
indium-1.0.33+mc1.21.jar
journeymap-fabric-1.21-6.0.0-beta.1.jar
modmenu-11.0.0-rc.4.jar
MouseTweaks-fabric-mc1.21-2.26.jar
placeholder-api-2.4.0-pre.2+1.21.jar
quickrightclick-1.21.0-1.4.jar
reeses_sodium_options-1.7.2+mc1.21.jar
remove-terralith-intro-message-1.0.jar
skinlayers3d-fabric-1.6.6-mc1.21.jar
sodium-fabric-0.5.9+mc1.21.jar
spark-1.10.73-fabric.jar
Terralith_1.21_v2.5.3.jar
viewdistancefix-fabric-1.21-1.0.2.jar
world-weaver-21.0.1.jar
wthit-fabric-12.2.0.jar
YetAnotherConfigLib-3.5.0+1.21-fabric.jar
Zoomify-2.14.0+1.21.jar
```

**Comment by DarkDevero** [comment]
[latest.txt](https://github.com/user-attachments/files/15908829/latest.txt)

**Comment by Druivenpasta** [comment]
![image](https://github.com/quiqueck/BCLib/assets/94719433/786ff77f-8b19-4edd-bb44-e50e6436ae7d)
All BetterX mods are still porting, you'll have to switch to 1.20 for a short while.

**Comment by KennyDevT** [comment]
> All BetterX mods are still porting, you'll have to switch to 1.20 for a short while.

They are released as Alpha so should still expect to atleast boot.

**Comment by StupidRepo** [comment]
> > All BetterX mods are still porting, you'll have to switch to 1.20 for a short while.
> 
> They are released as Alpha so should still expect to atleast boot.

Exactly. To be honest, I think it's slightly lazy to rush out a release (regardless of whether it's Alpha, Beta, or Stable) without at least testing if it **launches**. Also, should the dev(s) not have the time to test it, they shouldn't release it until they do!

**Comment by OpenBagTwo** [comment]
@StupidRepo No one in this thread is the mod's developer. And no one is forcing you to download the mod (which, btw, you have not paid a penny to support the development of).

The entire BetterX suite is maintained by one developer with a full-time job (note: it's not me either), and if you download an Alpha release, it is entirely reasonable that you should expect to be treated as an "early access" tester.

**Comment by StupidRepo** [comment]
> @StupidRepo No one in this thread is the mod's developer. And no one is forcing you to download the mod (which, btw, you have not paid a penny to support the development of).
> 
> The entire BetterX suite is maintained by one developer with a full-time job (note: it's not me either), and if you download an Alpha release, it is entirely reasonable that you should expect to be treated as an "early access" tester.

1. Never said anyone here was the mod's developer. What's your point?
2. I don't see a valid reason why I should need to "support" the development. Here's three questions for you (and 2 more below): How does my money help to support the development? Why should I pay money to support a mod for a game? Why does anyone need any of my money to support development?
3. Why make a mod, hell even multiple mods, when you don't even have the time to support them? That's a bit silly, no?

Simply put, this is a very pointless argument (at least to me it is). Whilst no one is stopping you from responding to my questions, don't expect me to respond any further.

**Comment by OpenBagTwo** [comment]
@StupidRepo My point is that you are acting _extremely_ entitled, while also directing your frustration and hostility at people who are in the same boat as you and only trying to help.

**Comment by Druivenpasta** [comment]
In Repo's defense, even if a mod is in alpha it's good to have intel on why the mod is crashing, so it can be prevented later. More logs have never hurt anyone. Let's deescalate this situation now :)

**Comment by ErzeKawek** [comment]
It Is Currently Crashing because Quiqueck Is Porting some parts of BCLib to WorldWeaver, so no wonder it still crashes, BCLib 21.0.0 is Heavily Unfinished.

**Comment by ErzeKawek** [comment]
> > @StupidRepo No one in this thread is the mod's developer. And no one is forcing you to download the mod (which, btw, you have not paid a penny to support the development of).
> > The entire BetterX suite is maintained by one developer with a full-time job (note: it's not me either), and if you download an Alpha release, it is entirely reasonable that you should expect to be treated as an "early access" tester.
> 
> 1. Never said anyone here was the mod's developer. What's your point?

Yes we aren't
> 2. I don't see a valid reason why I should need to "support" the development. Here's three questions for you (and 2 more below): How does my money help to support the development? Why should I pay money to support a mod for a game? Why does anyone need any of my money to support development?

Who Said You need to Support the Development?
> 3. Why make a mod, hell even multiple mods, when you don't even have the time to support them? That's a bit silly, no?

He Has Time To Support them, Just Look at the amount of Commits he made on the 1.21 Branch
> 
> Simply put, this is a very pointless argument (at least to me it is). Whilst no one is stopping you from responding to my questions, don't expect me to respond any further.

**Comment by ErzeKawek** [comment]
> > All BetterX mods are still porting, you'll have to switch to 1.20 for a short while.
> 
> They are released as Alpha so should still expect to atleast boot.

@KennyDevT 
Well it is heavily unfinished because of Code Changes In 1.21.
